shell/platform/darwin/macos/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -0,0 +1,87 @@
+  FlutterViewController* viewController = [[FlutterViewController alloc] initWithProject:project];
+
+  // Mock getPasteboard so that this test will work in environments without a
+  // real pasteboard.
+  id pasteboardMock = OCMClassMock([NSPasteboard class]);
+  __block NSString* clipboardString = @"";
+  void (^setStringMock)(NSInvocation*) = ^(NSInvocation *invocation) {
+    [invocation getArgument:&clipboardString atIndex:2];
+  };
+  OCMStub([pasteboardMock setString:[OCMArg any] forType:[OCMArg any]]).andDo(setStringMock);
+  void (^stringForTypeMock)(NSInvocation*) = ^(NSInvocation *invocation) {
+   [invocation setReturnValue:&clipboardString];
+  };
+  OCMExpect([pasteboardMock stringForType:[OCMArg any]]).andDo(stringForTypeMock);
+  id viewControllerMock = OCMPartialMock(viewController);
+  OCMStub([viewControllerMock getPasteboard]).andReturn(pasteboardMock);
+
+  // First call setClipboardData to put a string on the pasteboard.
+  __block bool calledSet = false;
+  FlutterResult resultSet = ^(id result) {
+    calledSet = true;
+  };
+  FlutterMethodCall* methodCallSet =
+      [FlutterMethodCall methodCallWithMethodName:@"Clipboard.setData"
+                                        arguments:@{@"text" : @"some string"}];
+  [viewControllerMock handleMethodCall:methodCallSet result:resultSet];
+  ASSERT_TRUE(calledSet);
+
+  // Call hasStrings and expect it to be true.
+  __block bool called = false;
+  __block bool value;
+  FlutterResult result = ^(id result) {
+    called = true;
+    NSNumber* valueNumber = [result valueForKey:@"value"];
+    value = [valueNumber boolValue];
+  };
+  FlutterMethodCall* methodCall =
+      [FlutterMethodCall methodCallWithMethodName:@"Clipboard.hasStrings" arguments:nil];
+  [viewControllerMock handleMethodCall:methodCall result:result];
+  ASSERT_TRUE(called);
+  ASSERT_TRUE(value);
+
+  // Now call setData again to clear the pasteboard.
+  __block bool calledSetClear = false;
+  FlutterResult resultSetClear = ^(id result) {
+    calledSetClear = true;
+  };
+  FlutterMethodCall* methodCallSetClear =
+      [FlutterMethodCall methodCallWithMethodName:@"Clipboard.setData"
+                                        arguments:@{@"text" : [NSNull null]}];
+  [viewControllerMock handleMethodCall:methodCallSetClear result:resultSetClear];
+  ASSERT_TRUE(calledSetClear);
+
+  // Call hasStrings and expect it to be false.
+  __block bool calledAfterClear = false;
+  __block bool valueAfterClear;
+  FlutterResult resultAfterClear = ^(id result) {
+    calledAfterClear = true;
+    NSNumber* valueNumber = [result valueForKey:@"value"];
+    valueAfterClear = [valueNumber boolValue];
+  };
+  FlutterMethodCall* methodCallAfterClear =
+      [FlutterMethodCall methodCallWithMethodName:@"Clipboard.hasStrings" arguments:nil];
+  [viewControllerMock handleMethodCall:methodCallAfterClear result:resultAfterClear];
+  ASSERT_TRUE(calledAfterClear);
+  ASSERT_FALSE(valueAfterClear);
+}
```

Comment:
This test is doing a lot of different things; please make individual tests that are more targeted. You're already using a fake pasteboard, so there's no need to do the entire cycle in a single test.

Reply:
Good call, I split it into two and it's a lot cleaner.